cardano-cli/src/Cardano/CLI/Run.hs

```diff
@@ -2,32 +2,54 @@
 -- | Dispatch for running all the CLI commands
 module Cardano.CLI.Run
   ( ClientCommand(..)
+  , ClientCommandErrors
+  , renderClientCommandError
   , runClientCommand
   ) where
 
 import           Cardano.Prelude
 
 import qualified Data.Text as Text
 import           Control.Monad.Trans.Except (ExceptT)
+import           Control.Monad.Trans.Except.Extra (firstExceptT)
 
-import           Cardano.CLI.Errors (CliError(..))
-import           Cardano.CLI.Commands (ClientCommand(..))
-import           Cardano.CLI.Byron.Run (runByronClientCommand)
-import           Cardano.CLI.Shelley.Run (runShelleyClientCommand)
+import           Cardano.CLI.Byron.Commands (ByronCommand)
+import           Cardano.CLI.Byron.Run (ByronClientCmdError, runByronClientCommand)
+import           Cardano.CLI.Shelley.Commands (ShelleyCommand)
+import           Cardano.CLI.Shelley.Run (ShelleyClientCmdError, runShelleyClientCommand)
 
 import           Data.Version (showVersion)
 import           Paths_cardano_cli (version)
 import           System.Info (arch, compilerName, compilerVersion, os)
 import           Cardano.Config.GitRev (gitRev)
 
+-- | Sub-commands of 'cardano-cli'.
+data ClientCommand =
 
-runClientCommand :: ClientCommand -> ExceptT CliError IO ()
-runClientCommand (ByronCommand   c) = runByronClientCommand c
-runClientCommand (ShelleyCommand c) = runShelleyClientCommand c
+    -- | Byron Related Commands
+    ByronCommand ByronCommand
+
+    -- | Shelley Related Commands
+  | ShelleyCommand ShelleyCommand
+
+  | DisplayVersion
+  deriving Show
```

Comment:
I'm not sure that merging this into this module, from the `Cardano.CLI.Commands` is needed. We still have the separate `Cardano.CLI.Byron.Commands` and `Cardano.CLI.ShelleyCommands`, and it's quite nice to follow the same pattern.

Why have the command types separate? So that the CLI parsers only depend on that command types module and not also all of the run actions.

Whereas the error types are fundamentally related to the run actions, so they certainly live here.